test/helpers/search_helper_test.rb

```diff
@@ -0,0 +1,19 @@
+require 'test_helper'
+
+class RecordHelperTest < ActionView::TestCase
+  include SearchHelper
+
+  test 'removes displayed fields from highlights' do
+    result = { 'highlight' => [{ 'matchedField' => 'title', 'matchedPhrases' => 'Very important data' },
+                               { 'matchedField' => 'title.exact_value', 'matchedPhrases' => 'Very important data' },
+                               { 'matchedField' => 'content_type', 'matchedPhrases' => 'Dataset' },
+                               { 'matchedField' => 'dates.value', 'matchedPhrases' => '2022' },
+                               { 'matchedField' => 'contributors.value', 'matchedPhrases' => 'Jane Datascientist' }] }
+    assert_empty trim_highlights(result)
+  end
+
+  test 'does not remove undisplayed fields from highlights' do
+    result = { 'highlight' => [{ 'matchedField' => 'summary', 'matchedPhrases' => 'Have some data' }] }
+    assert_equal [{ 'matchedField' => 'summary', 'matchedPhrases' => 'Have some data' }], trim_highlights(result)
+  end
+end
```

Comment:
This is non-blocking, but it might be useful to have a test that shows when the highlights contain both displayed and undisplayed fields, `trim_highlights` returns just the undisplayed fields. That is mostly shown in these tests but since they only cover the two edges, a test for the middle (i.e. both types of fields in the result) would make me feel a little more confident if we make changes to the helper in the future.